script.js

```diff
@@ -0,0 +1,201 @@
+
+window.addEventListener("DOMContentLoaded", () => {
+  updateCnt();
+  let items = getLocalStorage();
+
+  if (items.length > 0) {
+    items.forEach(function (item) {
+      if (item.value.type === "todo") {
+        createTodoItem(item.id, item.value.text);
+      } else if (item.value.type === "done") {
+        createDoneItem(item.id, item.value.text);
+      }
+    });
+  }
+
+  // LocalStorage에서 Item을 불러오고, 비어있는 경우 예외
+});
+
+// ===== 등록 이벤트핸들러, Item 수 추적 ===== //
+
+form.addEventListener("submit", addTodo);
+
+function updateCnt() {
+  todoCountField.textContent = todoList.children.length;
+  doneCountField.textContent = doneList.children.length;
+}
+
+// ====== Todo, Done List 조작 관련 함수들 ======= //
+
+function addTodo(e) {
+  e.preventDefault();
+  const value = formInput.value;
+  const id = new Date().getTime().toString();
+
+  if (value !== "") {
+    createTodoItem(id, value);
+    addToLocalStorage(id, { text: value, type: "todo" });
+  }
+
+  formInput.value = "";
+  updateCnt();
+
+  // Form에서 넘어온 값이 공백인 경우 넘기기
+  // Unique ID를 생성하기 위해 등록시점의 시간을 사용 (millisecond 단위)
+}
+
+function moveToDone(e) {
+  const todoItem = e.currentTarget.parentElement;
+  const id = todoItem.getAttribute("id");
+  const itemText = todoItem.textContent.slice(0, -3);
+
+  todoList.removeChild(todoItem);
+  editLocalStorage(id, { text: itemText, type: "done" });
+
+  createDoneItem(id, itemText);
+  updateCnt();
+
+  // button 기준 parentELement = doneItem
+  // Unique ID를 기준으로 localStorage 수정
+}
+function moveToTodo(e) {
+  const doneItem = e.currentTarget.parentElement;
+  const id = doneItem.getAttribute("id");
+  const itemText = doneItem.textContent.slice(0, -3);
+
+  //textContent로 불러올 시 "  \n"이 더해지는 현상이 있어 slice로 처리
+
+  doneList.removeChild(doneItem);
+  editLocalStorage(id, { text: itemText, type: "todo" });
+
+  createTodoItem(id, itemText);
+  updateCnt();
+
+  // button 기준 parentELement = todoItem
+  // Unique ID를 기준으로 localStorage 수정
+}
+
+// ====== Item 조작 관련 함수들 ======= //
+
+function deleteTodoItem(e) {
+  const todoItem = e.currentTarget.parentElement;
+  const id = todoItem.getAttribute("id");
+  todoList.removeChild(todoItem);
+  removeFromLocalStorage(id);
+  updateCnt();
+
+  // button 기준 parentELement = doneItem
+}
+
+function deleteDoneItem(e) {
+  const doneItem = e.currentTarget.parentElement;
+  const id = doneItem.getAttribute("id");
+  doneList.removeChild(doneItem);
+  removeFromLocalStorage(id);
+  updateCnt();
+
+  // button 기준 parentELement = todoItem
+}
+
+function createTodoItem(id, itemText) {
+  const todoItem = document.createElement("div");
+  let attribute = document.createAttribute("id");
+  attribute.value = id;
+  todoItem.setAttributeNode(attribute);
+  todoItem.classList.add("item");
+
+  // 새로운 div element 생성 후 인자로 전달된 unique id를 attribute으로 설정
+  // 'item' 클래스는 각 아이템의 css 스타일 프리셋
+
+  todoItem.innerHTML = `<button class="itemFinish" type="button"><p class="itemTitle">${itemText}</p></button>
```

Comment:
```suggestion
  todoItem.innerHTML = `
  <button class="itemFinish" type="button">
      <p class="itemTitle">${itemText}</p>
  </button>`
```

템플릿 리터럴은 줄바꿈과 들여쓰기도 저장하므로 컴포넌트의 구조가 보일 수 있도록 쓰는게 좋겠죠.

Reply:
감사합니다. 해당 부분 참고해서 리터럴 작성하도록 하겠습니다!